infertrade/algos/community/allocations.py

```diff
@@ -24,18 +24,49 @@
 from infertrade.PandasEnum import PandasEnum
 import infertrade.algos.community.signals
 
+
+def add_allocation_options(allocation_function: callable) -> callable:
+    def function_with_options(*args, **kwargs) -> pd.DataFrame:
+        allocation_upper_limit = allocation_lower_limit = None
+        if 'allocation_upper_limit' in kwargs:
+            allocation_upper_limit = kwargs.pop('allocation_upper_limit')
+            if not isinstance(allocation_upper_limit, (int, float)):
+                raise TypeError('allocation_upper_limit, if assigned, must be an int or a float.')
+        if 'allocation_lower_limit' in kwargs:
+            allocation_lower_limit = kwargs.pop('allocation_lower_limit')
+            if not isinstance(allocation_lower_limit, (int, float)):
+                raise TypeError('allocation_lower_limit, if assigned, must be an int or a float.')
+        dataframe = allocation_function(*args, **kwargs)
+        if allocation_upper_limit:
+            dataframe.loc[
+                    dataframe[PandasEnum.ALLOCATION.value]
+                    > allocation_upper_limit, PandasEnum.ALLOCATION.value
+                    ] = allocation_upper_limit
+        if allocation_lower_limit:
+            dataframe.loc[
+                    dataframe[PandasEnum.ALLOCATION.value]
+                    <
+                    allocation_lower_limit, PandasEnum.ALLOCATION.value
+                    ] = allocation_lower_limit
+        return dataframe
+    return function_with_options
+
```

Comment:
We could maybe more the decorator to the operations utilities file, as it is an extension to the algorithm calculations.